pkg/controller/controller_utils.go

```diff
@@ -1110,10 +1110,14 @@ func getOrCreateServiceAccount(coreClient v1core.CoreV1Interface, namespace, nam
 
 	// Create the namespace if we can't verify it exists.
 	// Tolerate errors, since we don't know whether this component has namespace creation permissions.
-	if _, err := coreClient.Namespaces().Get(namespace, metav1.GetOptions{}); apierrors.IsNotFound(err) {
+	_, err = coreClient.Namespaces().Get(namespace, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
 		if _, err = coreClient.Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}); err != nil && !apierrors.IsAlreadyExists(err) {
 			klog.Warningf("create non-exist namespace %s failed:%v", namespace, err)
 		}
+	} else if err != nil {
```

Comment:
Hmm, this seems to be in contrast to the comment above which says `tolerate errors, since we don't know whether this component has namespace creation permissions`.

Reply:
I modified the condition to be clearer.
I don't think there is contradiction since the above comment is about the warning on line 1116 (which is not Errorf) w.r.t. the re-assigned err.
My check is about the initial err.

@mattjmcnaughton 
Can you take another look ?